### PR TITLE
Update Diagnostic category and ID ranges to reflect currently used IDs

### PR DESCRIPTION
**Pull request by mavasani**
The category to ID ranges in `DiagnosticCategoryAndIdRanges.txt` file currently reflect the **reserved** ID ranges for the category. This change updates the CA ranges to now reflect the **currently used** ID ranges for the category. This makes it easier to identify the next available ID for a specific CA category when implementing a new rule. There will be an additional step of updating the range end in this text file when implementing a new rule, but that seems a good tradeoff versus spending additional time figuring out the correct next available ID to use. I will send a separate PR to add a code fix to automatically update this text file.

**NOTE:** The category to ID ranges in `DiagnosticCategoryAndIdRanges.txt` text file are enforced via an analyzer, so you will see warnings/build breaks if you choose a rule ID with incorrect category and/or choose a rule ID outside the specified range for a category.

### CODE REVIEW

**Comment by mavasani** [comment]
Tag @stephentoub @bartonjs @pgovind @buyaa-n @jeffhandley - the goal here is to make it easier for you folks to identify the next available CA rule ID in the selected rule category. Let me know if this sounds good to you, and I can add a code fix to update the category-ID range file when adding a new rule.

**Comment by mavasani** [comment]
Also tagging @Evangelink as this will affect your PRs as well..

**Comment by Evangelink** [comment]
@mavasani That's a really good idea, I was considering doing something similar to make it easier to find the next available ID!

File: src/Utilities/Compiler/DiagnosticCategoryAndIdRanges.txt [hunk 1]
```diff
@@ -2,17 +2,23 @@
...
+# The ranges below represent the **currently used IDs** for the corresponding category.
+# When implementing a new rule for the category:
+#   1. Choose the rule ID immediately following the range end.
+#   2. Update the range end to the chosen rule ID.
+#
+Design: CA2210, CA1000-CA1069
+Globalization: CA2101, CA1300-CA1309
+Mobility: CA1600-CA1601
+Performance: HA, CA1800-CA1829
+Security: CA2100-CA2153, CA2300-CA2399, CA3000-CA3147, CA5300-CA5499
```
Comment (mavasani) [inline src/Utilities/Compiler/DiagnosticCategoryAndIdRanges.txt]: @dotpaul Let me know if you'd like to make a similar change to switch from reserved IDs to currently used IDs. I can create a separate PR on 2.9.x branch just for security rules.

Comment (dotpaul) [reply src/Utilities/Compiler/DiagnosticCategoryAndIdRanges.txt]: Sure, that'd be great

Comment (mavasani) [inline src/Utilities/Compiler/DiagnosticCategoryAndIdRanges.txt]:
```suggestion
Security: CA2100-CA2153, CA2300-CA2330, CA3000-CA3147, CA5300-CA5403
```

Comment (mavasani) [reply src/Utilities/Compiler/DiagnosticCategoryAndIdRanges.txt]: @dotpaul I have updated the ranges and retargeted this PR to 2.9.x branch.

**Comment by stephentoub** [comment]
> Let me know if this sounds good to you, and I can add a code fix to update the category-ID range file when adding a new rule.

Thanks, @mavasani.  Seems reasonable.

**Pull request merged**